pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ExpressionTransformer.java

```diff
@@ -132,6 +132,7 @@ public GenericRow transform(GenericRow record) {
             record.putValue(column, transformFunctionEvaluator.evaluate(record));
           } catch (Exception e) {
             record.putValue(column, null);
+            record.putValue(GenericRow.INCOMPLETE_RECORD_KEY, true);
```

Comment:
Let's make this consistent with other transformers and move the if check into the catch block, and log a debug info. `record.putValue(column, null);` is redundant